ganttproject/src/net/sourceforge/ganttproject/search/TaskSearchService.java

```diff
@@ -42,20 +42,33 @@ public TaskSearchService() {
   public List<MySearchResult> search(String query) {
     query = query.toLowerCase();
     List<MySearchResult> results = new ArrayList<MySearchResult>();
+    String snippet = "";
+    String snippetText = "";
     for (Task t : getProject().getTaskManager().getTasks()) {
       boolean matched = false;
+      if (isNotEmptyAndContains(t.getName(), query)) {
+        matched = true;
+      }
       for (CustomProperty c : t.getCustomValues().getCustomProperties()) {
         if (isNotEmptyAndContains(c.getValueAsString(), query)) {
           matched = true;
+          snippet = c.getDefinition().getName();
+          snippetText = c.getValueAsString();
           break;
         }
       }
-      if (isNotEmptyAndContains(t.getName(), query) || isNotEmptyAndContains(t.getNotes(), query)
-          || isNotEmptyAndContains(String.valueOf(t.getTaskID()), query)) {
+      if (isNotEmptyAndContains(t.getNotes(), query)) {
+        matched = true;
+        snippet = "Notes";
+        snippetText = t.getNotes();
```

Comment:
If a note is super long it is possible for the text to go on and on. Wasn't sure how to tackle that. Matching a keyword on text and having ellipses around complete words so many characters to the right and left opens up a whole ton of fringe cases.